File: 백준/Bronze/10810. 공 넣기/공 넣기.java
import java.util.Scanner;
public class Main{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int[] arr = new int[in.nextInt()];
        int n = in.nextInt();
        for(int i = 1; i <= n; i++){
            int a = in.nextInt()-1, b = in.nextInt()-1;
            int v = in.nextInt();
            for(int j = a; j <= b; j++){
                arr[j] = v;
            }
        }
        for(int i = 0; i < arr.length; i++){
            System.out.printf("%d ",arr[i]);
        }
    }
}

File: 백준/Bronze/10810. 공 넣기/공 넣기.kt
import java.util.Scanner
fun main(){
    val sc = Scanner(System.`in`)
    val arr = IntArray(sc.nextInt(), {0})
    for(i in 1..sc.nextInt()){
        var a = sc.nextInt()-1; var b = sc.nextInt()-1
        var n = sc.nextInt()
        for(j in a..b)
            arr[j] = n
    }
    arr.forEach { print("$it ") }
}